### DISCUSSION TITLE
Instantiating a freejoint

### DISCUSSION THREAD

**Discussion by abhijeetknayak** [google-deepmind/mujoco#2041] [Asking for Help] [answered]
Hey there, I would like to create dynamic objects in the mujoco xml file:

    <body name="dynamic_body" pos="0 0 1">
        <joint type="free" name="free_joint"/>
        <geom type="sphere" size="0.1" rgba="1 0 0 1"/>
    </body>

However, this gives me the following error while reading the XML: **ValueError: Error: key 0: invalid qpos size, expected length 16** 
Could you help me understand what I'm doing wrong?

**Comment by yuvaltassa** [comment, marked as answer]
Is the error not clear? you model has a keyframe that is of the wrong size.
I guess we should replace "key" with "keyframe" to make this clearer

**Reply by abhijeetknayak** [reply]
Thank you for your reply! Yeah, my bad. I forgot that I had to update the keyframe after adding a new joint.
